### PR TITLE
[chore] Add OpenTelemetry Demo for auto-instrumentation example

### PR DESCRIPTION
**Pull request by jvoravong**
**Description:**
- Adding a new demo that we can use to show case how to auto-instrument Java, .Net, Python, NodeJs, and more.
- This demo is a filtered derivative of https://raw.githubusercontent.com/open-telemetry/opentelemetry-demo/main/kubernetes/opentelemetry-demo.yaml
  - We have to strip away some of the env vars and deployments  
  - The [otel-demo.yaml](https://github.com/signalfx/splunk-otel-collector-chart/pull/954/files#diff-b73494f85d622ff1f9d971a30dac724e4ddf6937a8f2c875cabd28004d6f3742) added in this PR is auto-generated, so reviewers are advised not to inspect this file in great depth.
- Will be adding related documentation on how to use this demo with various instrumentation libraries soon. 
 
**Testing:** Manually tested locally.

### CODE REVIEW

File: examples/enable-operator-and-auto-instrumentation/otel-demo/otel-demo.yaml [hunk 1]
```diff
@@ -0,0 +1,7523 @@
...
+kind: ConfigMap
+metadata:
+  name: opentelemetry-demo-grafana
+  namespace: otel-demo
+  labels:
+    app.kubernetes.io/name: grafana
+    app.kubernetes.io/instance: opentelemetry-demo
+    app.kubernetes.io/version: "10.0.3"
+data:
+  grafana.ini: |
```
Comment (atoulme) [inline examples/enable-operator-and-auto-instrumentation/otel-demo/otel-demo.yaml:80]: can we remove grafana from the demo?

Comment (jvoravong) [reply examples/enable-operator-and-auto-instrumentation/otel-demo/otel-demo.yaml:80]: I'll see if we can remove grafana and prometheus, should be possible.

Comment (jvoravong) [reply examples/enable-operator-and-auto-instrumentation/otel-demo/otel-demo.yaml]: Removed grafana, prometheus, and jaeger from otel-demo.yaml.

**Review by atoulme — APPROVED** [review]
LGTM

**Pull request merged**